rclpy/src/rclpy/wait_set.cpp

```diff
@@ -318,23 +283,62 @@ get_ready_entities(const std::string & entity_type, py::capsule pywait_set)
 }
 
 void
-wait(py::capsule pywait_set, int64_t timeout)
+WaitSet::wait(int64_t timeout)
 {
-  if (0 != std::strcmp("rcl_wait_set_t", pywait_set.name())) {
-    throw py::value_error("capsule is not an rcl_wait_set_t");
-  }
-  auto wait_set = static_cast<rcl_wait_set_t *>(pywait_set);
-
   rcl_ret_t ret;
 
   // Could be a long wait, release the GIL
   {
     py::gil_scoped_release gil_release;
-    ret = rcl_wait(wait_set, timeout);
+    ret = rcl_wait(rcl_wait_set_.get(), timeout);
   }
 
-  if (ret != RCL_RET_OK && ret != RCL_RET_TIMEOUT) {
+  if (RCL_RET_OK != ret && RCL_RET_TIMEOUT != ret) {
     throw RCLError("failed to wait on wait set");
   }
 }
+void define_waitset(py::object module)
+{
+  py::class_<WaitSet, Destroyable, std::shared_ptr<WaitSet>>(module, "WaitSet")
+  .def(py::init<size_t, size_t, size_t, size_t, size_t, size_t, py::capsule>())
+  .def(py::init<>())
+  .def_property_readonly(
+    "pointer", [](const WaitSet & waitset) {
+      return reinterpret_cast<size_t>(waitset.rcl_ptr());
+    },
+    "Get the address of the entity as an integer")
+  .def(
+    "wait_set_clear_entities", &WaitSet::wait_set_clear_entities,
+    "Clear all the pointers in the wait set")
+  .def(
+    "wait_set_add_entity", &WaitSet::wait_set_add_entity,
+    "Add an entity to the wait set structure")
+  .def(
+    "wait_set_add_service", &WaitSet::wait_set_add_service,
+    "Add a service to the wait set structure")
+  .def(
+    "wait_set_add_subscription", &WaitSet::wait_set_add_subscription,
+    "Add a subcription to the wait set structure")
+  .def(
+    "wait_set_add_client", &WaitSet::wait_set_add_client,
+    "Add a client to the wait set structure")
+  .def(
+    "wait_set_add_timer", &WaitSet::wait_set_add_timer,
+    "Add a timer to the wait set structure")
+  .def(
+    "wait_set_clear_entities", &WaitSet::wait_set_clear_entities,
+    "Add a client to the wait set structure")
+  .def(
+    "wait_set_add_event", &WaitSet::wait_set_add_event,
+    "Add an event to the wait set structure")
+  .def(
+    "wait_set_is_ready", &WaitSet::wait_set_is_ready,
+    "Check if an entity in the wait set is ready by its index")
+  .def(
+    "get_ready_entities", &WaitSet::get_ready_entities,
+    "Get list of non-null entities in wait set")
```

Comment:
@ahcorde nit: perhaps
```suggestion
    "Get list of entities ready by entity type")
```
would be more accurate?